includes/Product/Hooks.php

```diff
@@ -19,6 +19,180 @@ class Hooks {
+                foreach ( $query_results as $result ) {
+                    $product    = wc_get_product( $result->ID );
+                    $price      = wc_price( $product->get_price() );
+                    $price_sale = $product->get_sale_price();
+                    $stock      = $product->get_stock_status();
+                    $sku        = $product->get_sku();
+                    $categories = wp_get_post_terms( $result->ID, 'product_cat' );
+
+                    if ( 'variable' === $product->get_type() ) {
+                        $price = wc_price( $product->get_variation_price() ) . ' - ' . wc_price( $product->get_variation_price( 'max' ) );
+                    }
+
+                    $get_product_image = esc_url( get_the_post_thumbnail_url( $result->ID, 'thumbnail' ) );
+
+                    if ( empty( $get_product_image ) && function_exists( 'wc_placeholder_img_src' ) ) {
+                        $get_product_image = wc_placeholder_img_src();
+                    }
+
+                    $output .= '<li>';
+                    $output .= '<a href="' . get_post_permalink( $result->ID ) . '">';
+                    $output .= '<div class="dokan-ls-product-image">';
+                    $output .= '<img src="' . $get_product_image . '">';
+                    $output .= '</div>';
+                    $output .= '<div class="dokan-ls-product-data">';
+                    $output .= '<h3>' . $result->post_title . '</h3>';
+
+                    if ( ! empty( $price ) ) {
+                        $output .= '<div class="product-price">';
+                        $output .= '<span class="dokan-ls-regular-price">' . $price . '</span>';
+                        if ( ! empty( $price_sale ) ) {
+                            $output .= '<span class="dokan-ls-sale-price">' . wc_price( $price_sale ) . '</span>';
+                        }
+                        $output .= '</div>';
+                    }
+
+                    if ( ! empty( $categories ) ) {
+                        $output .= '<div class="dokan-ls-product-categories">';
+                        foreach ( $categories as $category ) {
+                            if ( $category->parent ) {
+                                $parent  = get_term_by( 'id', $category->parent, 'product_cat' );
+                                $output .= '<span>' . $parent->name . '</span>';
+                            }
+                            $output .= '<span>' . $category->name . '</span>';
+                        }
+                        $output .= '</div>';
+                    }
+
+                    if ( ! empty( $sku ) ) {
+                        $output .= '<div class="dokan-ls-product-sku">' . esc_html__( 'SKU:', 'dokan-lite' ) . ' ' . $sku . '</div>';
+                    }
+
+                    $output .= '</div>';
+                    $output .= '</a>';
+                    $output .= '</li>';
+                }
+            }
+        }
+
+        if ( $output ) {
```

Comment:
cache the output, coz there are lots of things happening here, also delete the cache, if vendor update any product